gensim/models/coherencemodel.py

```diff
@@ -441,11 +442,16 @@ def topics(self, topics):
         self._topics = new_topics
 
     def _ensure_elements_are_ids(self, topic):
-        try:
-            return np.array([self.dictionary.token2id[token] for token in topic])
-        except KeyError:  # might be a list of token ids already, but let's verify all in dict
-            topic = (self.dictionary.id2token[_id] for _id in topic)
-            return np.array([self.dictionary.token2id[token] for token in topic])
+        elements_are_tokens = np.array([self.dictionary.token2id[token]
```

Comment:
Hanging indent please. Alternatively, break the list comprehension up to be more readable:

```python
tokens = [t for t in topic if token in self.dictionary.token2id]
elements_are_tokens = np.array([self.dictionary.token2id[t] for t in tokens])
...
```